monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -1,68 +1,37 @@
 import functools
-import json
 import logging
 from pprint import pformat
 from typing import List, Sequence
 
-import requests
-from requests.adapters import HTTPAdapter
-from urllib3 import Retry
-
 from common import AgentRegistrationData, AgentSignals, OperatingSystem
 from common.agent_configuration import AgentConfiguration
 from common.agent_event_serializers import AgentEventSerializerRegistry
 from common.agent_events import AbstractAgentEvent
-from common.common_consts.timeouts import MEDIUM_REQUEST_TIMEOUT, SHORT_REQUEST_TIMEOUT
+from common.common_consts.timeouts import SHORT_REQUEST_TIMEOUT
 from common.credentials import Credentials
 from common.types import AgentID, JSONSerializable, PluginType, SocketAddress
 
 from . import (
     AbstractIslandAPIClientFactory,
     IIslandAPIClient,
-    IslandAPIConnectionError,
     IslandAPIError,
     IslandAPIRequestError,
-    IslandAPIRequestFailedError,
-    IslandAPITimeoutError,
 )
+from .http_client import HTTPClient
+from .island_api_client_errors import IslandAPIResponseParsingError
 
 logger = logging.getLogger(__name__)
 
-# Retries improve reliability and slightly mitigates performance issues
-RETRIES = 5
-
 
-def handle_island_errors(fn):
-    @functools.wraps(fn)
-    def decorated(*args, **kwargs):
-        try:
-            return fn(*args, **kwargs)
-        except IslandAPIError as err:
-            raise err
-        except (requests.exceptions.ConnectionError, requests.exceptions.TooManyRedirects) as err:
-            raise IslandAPIConnectionError(err)
-        except requests.exceptions.HTTPError as err:
-            if 400 <= err.response.status_code < 500:
-                raise IslandAPIRequestError(err)
-            elif 500 <= err.response.status_code < 600:
-                raise IslandAPIRequestFailedError(err)
-            else:
-                raise IslandAPIError(err)
-        except TimeoutError as err:
-            raise IslandAPITimeoutError(err)
-        except Exception as err:
-            raise IslandAPIError(err)
-
-    return decorated
-
-
-def convert_json_error_to_island_api_error(fn):
+def handle_response_parsing_errors(fn):
     @functools.wraps(fn)
     def wrapper(*args, **kwargs):
         try:
             return fn(*args, **kwargs)
-        except (requests.JSONDecodeError, json.JSONDecodeError) as err:
-            raise IslandAPIRequestFailedError(err)
+        except IslandAPIError:
+            raise
```

Comment:
This is just for readability? It seems out of place under `handle_response_parsing_errors`. I'd say remove it. These errors are handled and raised where they're relevant.